packages/client-readmodel-writer/src/clientConsumerServiceV1.ts

```diff
@@ -1,25 +1,150 @@
 import { ClientCollection } from "pagopa-interop-commons";
-import { AuthorizationEventEnvelopeV1 } from "pagopa-interop-models";
+import {
+  AuthorizationEventEnvelopeV1,
+  fromClientV1,
+  fromKeyV1,
+  toReadModelClient,
+  toReadModelKey,
+} from "pagopa-interop-models";
 import { match } from "ts-pattern";
 
 export async function handleMessageV1(
   message: AuthorizationEventEnvelopeV1,
-  _clients: ClientCollection
+  clients: ClientCollection
 ): Promise<void> {
-  match(message)
+  await match(message)
     .with(
-      { type: "KeysAdded" },
-      { type: "KeyDeleted" },
-      { type: "KeyRelationshipToUserMigrated" },
       { type: "ClientAdded" },
-      { type: "ClientDeleted" },
       { type: "RelationshipAdded" },
-      { type: "RelationshipRemoved" },
       { type: "UserAdded" },
       { type: "UserRemoved" },
-      { type: "ClientPurposeAdded" },
-      { type: "ClientPurposeRemoved" },
-      () => Promise.resolve
+      async (message) => {
+        await clients.updateOne(
+          {
+            "data.id": message.stream_id,
+            "metadata.version": { $lt: message.version },
+          },
+          {
+            $set: {
+              data: message.data.client
+                ? toReadModelClient(fromClientV1(message.data.client))
+                : undefined,
+              metadata: {
+                version: message.version,
+              },
+            },
```

Comment:
Please double-check the Scala implementation to see if the updates are performed like so, or instead the addition/removal of a user should happen at this level (e.g. the `client` object doesn't contain the updated array of `userIds`)

Reply:
nice catch! this should be a `pull` operation.
In the same way, for `UserAdded` and `RelationshipAdded` should be performed a `push` operation.